rio_tiler/reader.py

```diff
@@ -193,7 +193,7 @@ def part(
 
     out_height = height or vrt_height
     out_width = width or vrt_width
-    if padding > 0 and not is_aligned(src_dst, bounds, out_height, out_width):
+    if padding > 0 and not is_aligned(src_dst, bounds, out_height, out_width, dst_crs):
```

Comment:
small change to `_requested_tile_aligned_with_internal_tile` method to allow other CRS than WebMercator